src/applications/proxy-rewrite/proxy-rewrite-entry.jsx

```diff
@@ -108,4 +101,54 @@ document.addEventListener('DOMContentLoaded', _e => {
     addOverlayTriggers();
     addFocusBehaviorToCrisisLineModal();
   });
-});
+}
+
+function activateInjectedAssets() {
+  const observer = new MutationObserver(createMutationObserverCallback());
+  observer.observe(document, {
+    attributes: true,
+    childList: true,
+    subtree: true,
+  });
+
+  const commonStore = createCommonStore();
+  document.addEventListener('DOMContentLoaded', _e => {
+    activateHeaderFooter(observer);
+    mountReactComponents(commonStore);
+  });
+}
+
+function getProxyRewriteCookieValue(
+  cookies = document.cookie,
+  parseCookie = cookie.parse,
+) {
+  return parseCookie(cookies).proxyRewrite;
+}
+
+function getMatchedWhitelistItem(whitelist = proxyRewriteWhitelist) {
+  const { hostname, pathname } = window.location;
+
+  return whitelist.find(
+    whitelistItem =>
+      whitelistItem.hostname === hostname &&
+      pathname.startsWith(whitelistItem.pathnameBeginning),
+  );
+}
+
+function shouldActivateInjectedAssets(whitelistItem, proxyRewriteCookieValue) {
+  if (whitelistItem.cookieOnly) {
+    return proxyRewriteCookieValue;
+  }
+
+  return !!whitelistItem;
+}
+
+if (
+  shouldActivateInjectedAssets(
```

Comment:
most of the changes are just moving code around into named functions. This is where the new gating behavior begins.